src/StudyRegistration/StudyRegistrationRequestForm.tsx

```diff
@@ -0,0 +1,300 @@
+            createKayakoTicket(subject, fullName, email, contents, 21).then(() => setFormSubmissionStatus({ status: 'success' }),
+              (err) => setFormSubmissionStatus({ status: 'error', text: err.message }));
+          } else {
+            console.error('Requestor returns 201 but no request_id in payload body'); // shouldn't get here
+          }
+        } else if (status === 409) {
+          // there is already a request for this user on this study, display a message and disable the button
+          setFormSubmissionButtonDisabled(true);
+          setFormSubmissionStatus({ status: 'warning', text: 'There is already a pending request for this study/user combination, please wait while we are processing your request.' });
+        } else {
+          // something has gone wrong
+          setFormSubmissionStatus({ status: 'error', text: `Failed to create a request with error code: ${status}. Please try again later. If the error persists, please contact us for help.` });
+        }
+        setReqAccessRequestPending(false);
+      },
+    );
+  };
+
+  const onFinish = (values) => {
+    setReqAccessRequestPending(true);
+    handleRegisterFormSubmission(values);
+  };
+
+  const onReset = () => {
+    form.resetFields();
+  };
+
+  const onRadioChange = (e) => {
+    setRole(e.target.value);
+  };
+
+  if (formSubmissionStatus) {
+    return (
+      <div className='study-reg-container'>
+        <div className='study-reg-form-container'>
+          {(formSubmissionStatus.status === 'success') ? (
+            <Result
+              status={formSubmissionStatus.status}
+              title='Your access request has been submitted!'
+              subTitle='You will be notified when a decision has been made'
+              extra={[
+                <Link key='discovery' to={'/discovery'}>
+                  <Button>Go To Discovery Page</Button>
+                </Link>,
+              ]}
+            />
+          ) : (
+            <Result
+              status={formSubmissionStatus.status}
+              title='A problem has occurred during submitting the request!'
+              subTitle={formSubmissionStatus.text}
+              extra={[
+                <Button type='primary' key='close' onClick={() => { setFormSubmissionStatus(null); setReqAccessRequestPending(false); }}>
+                  Close
+                </Button>,
+              ]}
+            />
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className='study-reg-container'>
+      <div className='study-reg-form-container'>
+        <Form className='study-reg-form' {...layout} form={form} name='study-reg-request-form' onFinish={onFinish} validateMessages={validateMessages}>
+          <Divider plain>Study Registration Access Request</Divider>
+          <Typography style={{ textAlign: 'center' }}>
+            Please fill out this form to request and be approved for access to register your study with the HEAL Platform.
```

Comment:
may want to add something like all fields required

Reply:
done